### ISSUE TITLE
opam lint: Warn about unkown variables

### ISSUE THREAD

**Issue by kit-ty-kate** [ocaml/opam#4000]
Some packages use variables such as `pkg:var` or `success` or `with-test` in the `build` or `post-message` fields for example, but mistakes might happen and the wrong variable name might be used or there might be some syntax mistake. Would it be possible to check for those variables to see if they might be undefined and raise a warning in that case? They might be custom variables but in most cases they won't be and I think a warning can be useful.

**Comment by rjbou** [comment]
A warning can be definitely useful, but it would also seems _weird_ as opam permit to check if a variable is defined or not in [filters](https://opam.ocaml.org/doc/Manual.html#Filters) (`?` operator).

**Comment by dra27** [comment]
From a review perspective of repository PRs, I imagine it would be handy to have a way for Camelus to *list* undefined variables so that a person can check that the use looks sound, but I agree that they don't want to be warnings.

**Comment by rjbou** [comment]
Yes, I see the need from a review perspective. For the moment, all linting check of opam repository are done in opam, but some of them should be separated. Worth adding a way to add lint check via opam-lib. New ones can also be used also in `opam-publish`.

**Comment by rjbou** [comment]
_digging some issues_ related to #1801
